src/main/scala/org/deeplearning4j/scalphagozero/scoring/GameResult.scala

```diff
@@ -122,36 +122,26 @@ object GameResult {
     statusMap.toMap
   }
 
-  private[this] final val deltas = List((-1, 0), (1, 0), (0, -1), (0, 1))
-
-  // Is it possible to rewrite this in a tailrec function ?
-  // And/or use another mecanism to browse the board ?
-  // Why not use the `Point(...).neighbors` method ?
-  private def collectRegion(
-      startingPoint: Point,
-      board: GoBoard,
-      visited: Set[Point] = Set.empty
-  ): (List[Point], Set[Option[Player]]) =
-    if (visited.contains(startingPoint)) (List.empty, Set.empty)
-    else {
-      val here: Option[Player] = board.getPlayer(startingPoint)
-
-      val (allPoints, allBorders) =
-        deltas.foldLeft((ListBuffer(startingPoint), mutable.Set.empty[Option[Player]])) {
-          case ((pointsAcc, bordersAcc), (row, col)) =>
-            val nextPoint = Point(startingPoint.row + row, startingPoint.col + col)
-            if (board.isOnGrid(nextPoint)) {
-              val neighbor: Option[Player] = board.getPlayer(nextPoint)
-              if (neighbor == here) {
-                val (points, borders) = collectRegion(nextPoint, board, visited + startingPoint)
-                (pointsAcc ++ points, bordersAcc ++ borders)
-              } else {
-                (pointsAcc, bordersAcc + neighbor)
-              }
-            } else (pointsAcc, bordersAcc)
-        }
+  private def collectRegion(startingPoint: Point, board: GoBoard): (List[Point], Set[Player]) = {
+    val initialPlayer = board.getPlayer(startingPoint)
+
+    val visitedPlayers = mutable.Set[Player]()
+    val visitedPoints = ListBuffer[Point](startingPoint)
 
-      (allPoints.toList, allBorders.toSet)
+    val toVisitPoints = mutable.Stack[Point](startingPoint)
+    while (toVisitPoints.nonEmpty) {
+      val toVisit = toVisitPoints.pop()
+      val toVisitColor = board.getPlayer(toVisit)
+      toVisitColor.foreach(visitedPlayers += _)
```

Comment:
`toVisitPlayer` instead of `toVisitColor` ;)

Reply:
+1